src/Cardano/CoinSelection/LargestFirst.hs

```diff
@@ -20,77 +21,140 @@ import Cardano.Types
 import Control.Arrow
     ( left )
 import Control.Monad
-    ( foldM, when )
+    ( foldM )
 import Control.Monad.Trans.Except
     ( ExceptT (..), except, throwE )
 import Data.Functor
     ( ($>) )
 import Data.List.NonEmpty
     ( NonEmpty (..) )
 import Data.Ord
-    ( Down (..), comparing )
+    ( Down (..) )
 
 import qualified Data.List as L
 import qualified Data.List.NonEmpty as NE
 import qualified Data.Map.Strict as Map
 
--- | Implements the __Largest-First__ coin selection algorithm.
+-- | Generate a coin selection according to the __largest first__ algorithm.
+--
+-- For the given /output list/ and /starting UTxO set/, this algorithm
+-- generates a /coin selection/ that is capable of paying for all of the
+-- outputs, and a /remaining UTxO set/ from which spent values have been
+-- removed.
+--
+-- === Cardinality Rules
+--
+-- The algorithm requires that:
+--
+--  1.  Each output is paid for by /one or more/ entries from the UTxO set.
+--
+--  2.  Each entry from the UTxO set is used to pay for /at most one/ output.
+--
+--      (A given entry from the UTxO set /cannot/ be used to pay for multiple
+--      outputs.)
+--
+-- === Order of Processing
+--
+-- The algorithm processes /both/ the given output list /and/ the supplied UTxO
+-- set in __descending order of coin value__, from largest to smallest.
+--
+-- At all stages of processing, the algorithm maintains a /remaining UTxO set/
+-- that is steadily depleted as outputs are paid for.
+--
+-- === Processing an Output
+--
+-- For /each output/ in the (descending) output list, the algorithm repeatedly
+-- selects unspent values from the /remaining UTxO set/ (in descending order)
+-- until the /total selected value/ is greater than (or equal to) the output
+-- value, at which point the algorithm move ons to processing the /next/
+-- output.
+--
+-- If the /total selected value/ is greater than required for a particular
+-- output, the algorithm generates a /change output/ with the exact difference
+-- in value.
+--
+-- === Termination Conditions
+--
```

Comment:
Perhaps it would be worth adding the conditions under which successful termination occurs here.

Reply:
Fixed. See: https://github.com/input-output-hk/cardano-coin-selection/pull/7/commits/973126e81272aaef88f4875f7d6a75c5956ea95e